helix-core/src/test/java/org/apache/helix/integration/TestZkConnectionLost.java

```diff
@@ -103,6 +104,28 @@ public void afterClass() throws Exception {
     TestHelper.stopZkServer(_zkServerRef.get());
   }
 
+  @Test
+  public void testDisconnectWhenConnectionBreak() throws Exception {
+    _zkServerRef.get().shutdown();
+    AtomicBoolean disconnected = new AtomicBoolean(false);
+    Thread testThread = new Thread("Testing HelixManager disconnect") {
+      @Override
+      public void run() {
+        _controller.disconnect();
+        disconnected.set(true);
+      }
+    };
+    try {
+      testThread.start();
+      testThread.join(5000);
```

Comment:
Instead of giving a hard-coded timeout value, do you think we could use a CountDownLatch for this so the test doesn't depend on timing?

Reply:
I don't think so. Maybe I can make it a longer timeout. But if the logic is not working as expected, then the thread will never finish. So I'm not sure how the CountDownLatch can help here. Maybe you can elaborate on your idea?